corehq/apps/userreports/reports/view.py

```diff
@@ -97,6 +98,20 @@ def query_dict_to_dict(query_dict, domain):
     return request_dict
 
 
+def _has_location_filter(view_fn, *args, **kwargs):
+    """check that the report has at least one location choice provider filter
+    """
+    # God help me.
+    # We can't access the instance of the report view in the location_safe
+    # decorator
```

Comment:
nope, that's what I meant - the same logic, just in a different place.  Again though, I'm not convinced that'd be any clearer.